src/crystal/scheduler.cr

```diff
@@ -118,11 +136,25 @@ class Crystal::Scheduler
   end
 
   protected def yield : Nil
-    sleep(0.seconds)
+    {% if flag?(:unix) %}
+      sleep(0.seconds)
+    {% elsif flag?(:win32) %}
+      @runnables << @current
+      reschedule
+    {% else %}
+      {% raise "unsupported platform" %}
+    {% end %}
   end
 
   protected def yield(fiber : Fiber) : Nil
-    @current.resume_event.add(0.seconds)
+    {% if flag?(:unix) %}
+      @current.resume_event.add(0.seconds)
+    {% elsif flag?(:win32) %}
+      @runnables << fiber
```

Comment:
You can't push the current fiber into the runnables queue; it will lead to a situation where the current fiber will try to resume itself, for example.

You could, however, implement pseudos `Crystal::EventLoop` and `Crystal::Event` classes for win32 with a mere `Deque(Fiber)` and a Fiber that will pop & resume the next fiber from it. That would also avoid the unix/win32 checks everywhere (less noise), and have the API ready for implementing the event loop on win32.

Reply:
I suppose it would probably be better to remove these specifics from this PR.